_posts/2020-08-05-hive-acid-original-files.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title:  "Read support for original files of Hive transactional tables in Presto"
+author: Harmandeep Singh, Qubole
+excerpt_separator: <!--more-->
+---
+
+In Presto 331, read support for Hive transactional tables was introduced. It works well if a user creates a new Hive transactional table and reads it from Presto. However,  if an
+existing table is converted to a Hive transactional table, Presto would fail to read data from such a table because  read support for original files was missing. Original files are
+those files in a Hive transactional table that existed before the table was converted into a Hive transactional table and until version 340, Presto expected all files in a Hive
+transactional table to be in Hive ACID format. Users would have to perform a Major Compaction to convert original files into ACID files (i.e. base files) in such tables. This is
+not always possible as the original flat table (table in non-ACID format) could be huge and converting all the existing data into ACID format would be very expensive.
+
+This blog is an extension of the blog [Hive ACID and transactional tables' support in Presto](https://prestosql.io/blog/2020/06/01/hive-acid.html). It first describes original
+files and then goes into details of  read support for such files that was  added in Presto 340.
+
+<!--more-->
+
+# What are original files?
+
+Files present in flat ORC tables have the standard ORC schema. When a flat table is converted into a transactional table, existing files are not converted into Hive ACID format.
```

Comment:
Maybe instead of using "flat" table you can use non-transactional term instead?

Reply:
fixed.